game/scripts/vscripts/Sections/Chapter5/SectionRoshan.ts

```diff
@@ -0,0 +1,279 @@
+        chapter5Blockades.direMidTopRiver,
+        chapter5Blockades.midRiverTopSide,
+    ],
+};
+
+// Move to state when implementing final section of CH5
+const friendlyHeroesInfo = [
+    { name: "npc_dota_hero_tidehunter", loc: Vector(-2192, 1718, 0) }, // Offlane, tank
+    { name: "npc_dota_hero_juggernaut", loc: Vector(-2026, 1550, 0) }, // Carry
+    { name: "npc_dota_hero_mirana", loc: Vector(-2074, 1465, 0) }, // Position 4
+    { name: "npc_dota_hero_lion", loc: Vector(-2282, 1462, 0) }, // Position 5
+];
+
+const roshanMusic = "valve_ti10.music.roshan"
+
+function onStart(complete: () => void) {
+    print("Starting", sectionName);
+    const goalTracker = new GoalTracker();
+    const goalEnterRoshPit = goalTracker.addBoolean("Walk into Roshan's pit.");
+    const goalUpgradeTalents = goalTracker.addBoolean("Choose and pick between the available talents.")
+    const goalDefeatRoshan = goalTracker.addBoolean("Defeat Roshan!");
+    const goalPickupAegis = goalTracker.addBoolean("Pick up the Aegis of the Immortal.")
+    const goalLeaveRoshPit = goalTracker.addBoolean("Leave Roshan's pit and move to the next marker.")
+
+    const roshPitGoalPosition = Vector(-2600, 2200, 28)
+    const leaveRoshPitGoalPosition = Vector(-2140, 1740, 0)
+
+    const playerHero = getOrError(getPlayerHero())
+
+    const itemDaedalus = "item_greater_crit"
+    const itemAC = "item_assault"
+    const itemPowerTreads = "item_power_treads"
+    const itemHeart = "item_heart"
+    const itemAegis = "item_aegis"
+
+    let roshan = Entities.FindAllByName("npc_dota_roshan")[0] as CDOTA_BaseNPC
+
+    if (!roshan) {
+        roshan = CreateUnitByName("npc_dota_roshan", Vector(-2919, 2315, 32), true, undefined, undefined, DotaTeam.NEUTRALS)
+        roshan.FaceTowards(leaveRoshPitGoalPosition)
+        roshan.AddItemByName(itemAegis)
+    }
+
+    setupRoshanModifiers(roshan)
+
+    // Clear any Aegis boxes left on the ground
+    const droppedItems = Entities.FindAllByClassname("dota_item_drop") as CDOTA_Item_Physical[]
+
+    if (droppedItems) {
+        for (const droppedItem of droppedItems) {
+            const itemEntity = droppedItem.GetContainedItem()
+            if (itemEntity.GetAbilityName() === itemAegis) {
+                droppedItem.Destroy()
+            }
+        }
+    }
+
+    // DK lvl 25 talents
+    const dragonBlood25Talent = playerHero.FindAbilityByName("special_bonus_unique_dragon_knight")
+    const dragonTail25Talent = playerHero.FindAbilityByName("special_bonus_unique_dragon_knight_2")
+
+    graph = tg.withGoals(_ => goalTracker.getGoals(),
+        tg.seq([
+            tg.immediate(() => canPlayerIssueOrders = false),
+            tg.wait(1),
+            tg.immediate(() => setUnitPacifist(roshan, true)),
+            tg.fork([
+                tg.seq([
+                    tg.audioDialog(LocalizationKey.Script_5_Roshan_1, LocalizationKey.Script_5_Roshan_1, ctx => ctx[CustomNpcKeys.SlacksMudGolem], 3),
```

Comment:
The last argument here is the extra delay which can be removed here (default is 0.5s). You probably didn't notice this since it was bugged before but it was fixed now.